x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/configuration_form/source_field.tsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useState } from 'react';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiLink, EuiSpacer, EuiCallOut } from '@elastic/eui';
+
+import { documentationService } from '../../../../services/documentation';
+import {
+  UseField,
+  FormDataProvider,
+  FormRow,
+  ToggleField,
+  ComboBoxField,
+  FieldHook,
+} from '../../shared_imports';
+import { ComboBoxOption } from '../../types';
+
+const i18nTexts = {
```

Comment:
I like to have the i18n text out of the component as it clutters the JSX and it's harder to follow what is important. Here I can simply fold this `i18nTexts` in the IDE and focus on the component logic.